tests/linked_hash_set.rs

```diff
@@ -356,6 +356,7 @@ fn test_drain() {
 fn test_replace() {
     use core::hash;
 
+    #[allow(dead_code)]
```

Comment:
This looked intentional, so I figured it might make sense to suppress? Happy to change it to remove the unused field.

Reply:
Hmm, looking at it more I think the behavior is expected and suppressing the warning is correct?

`replace()` is documented as:

```rust
    /// Adds the given value to the set, replacing the existing value.
    ///
    /// If a previous value existed, returns the replaced value.  In this case, the value's position
    /// in the internal linked list is *not* changed.
```

and this is tested with:

```rust
    let mut s = LinkedHashSet::new();
    assert_eq!(s.replace(Foo("a", 1)), None);
    assert_eq!(s.len(), 1);
    assert_eq!(s.replace(Foo("a", 2)), Some(Foo("a", 1)));
    assert_eq!(s.len(), 1);

    let mut it = s.iter();
    assert_eq!(it.next(), Some(&Foo("a", 2)));
    assert_eq!(it.next(), None);
```

It seems to me that the test's intention is to test that the old value is not replaced here because old and new are identical (both in terms of `PartialEq` and in terms of `Hash`).

Reply:
Wait... isn't that what the method is supposed to do, replace the existing value and return the old one? Let me actually fully engage my brain and look at the test lol.

Reply:
Yeah, I think the test is supposed to look like this:

```rust
#[test]
fn test_replace() {
    use core::hash;

    #[derive(Debug)]
    struct Foo(&'static str, i32);

    impl PartialEq for Foo {
        fn eq(&self, other: &Self) -> bool {
            self.0 == other.0
        }
    }

    impl Eq for Foo {}

    impl hash::Hash for Foo {
        fn hash<H: hash::Hasher>(&self, h: &mut H) {
            self.0.hash(h);
        }
    }

    impl Foo {
        fn really_eq(&self, other: &Self) -> bool {
            self.0 == other.0 && self.1 == other.1
        }
    }

    let mut s = LinkedHashSet::new();
    assert_eq!(s.replace(Foo("a", 1)), None);
    assert_eq!(s.len(), 1);
    assert_eq!(
        s.replace(Foo("a", 2)).map(|f| f.really_eq(&Foo("a", 1))),
        Some(true)
    );
    assert_eq!(s.len(), 1);

    let mut it = s.iter();
    assert_eq!(it.next().map(|f| f.really_eq(&Foo("a", 2))), Some(true));
    assert_eq!(it.next(), None);
}
```

Reply:
Okay. Do you want to stick that change in a separate PR and merge that first?

Reply:
Uh, sure I can do that. That's actually probably the easiest thing to do, I just didn't want to step on your toes.

Reply:
Happy to rebase on top of your change!

Reply:
Rebased.